filewatcher/filewatcher.go

```diff
@@ -56,7 +62,28 @@ const (
 //
 // Please note that Parser should always return the consistent type.
 // Inconsistent type will cause panic, as does returning nil data and nil error.
-type Parser func(f io.Reader) (data interface{}, err error)
+type Parser func(f io.Reader) (data any, err error)
+
+// A DirParser is a callback function that will be called when the watched
+// directory has its content changed or is read for the first time.
+//
+// Please note that DirParser should always return the consistent type.
+// Inconsistent type will cause panic, as does returning nil data and nil error.
```

Comment:
```suggestion
// Please note that a DirParser must return a consistent type.
// Inconsistent types will cause a panic, as does returning nil data and nil error.
```

Why would a nil error cause a panic?

Reply:
That's the caveat from using [`atomic.Value`](https://pkg.go.dev/sync/atomic#Value.Store):

>Store sets the value of the Value to x. All calls to Store for a given Value must use values of the same concrete type. Store of an inconsistent type panics, as does Store(nil).

We can bypass this by doing something similar to 316e8443312ff705e8215947cb34297d81c1bc34, but I don't think we should do that unless absolutely necessary.